### PR TITLE
Update Media Composition Swagger to latest version

### PR DESCRIPTION
**Pull request by minnieliu**
# Data Plane API - Pull Request

## API Info: The Basics
Most of the information about your service should be captured in the issue that serves as your [*engagement record*](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/271/Azure-REST-API-Stewardship?anchor=rest-api-stewardship-process).

* Link to engagement record issue:

Is this review for (select one):

- [x] a private preview
- [ ] a public preview
- [ ] GA release

### Change Scope
<sup>This section will help us focus on the specific parts of your API that are new or have been modified. <br/>Please share a link to the design document for the new APIs, a link to the previous Open API document (swagger) if applicable, and the root paths that have been updated. </sup>
* Design Document:
* Previous Open API Doc:
* Updated paths:

## ❔Got questions? Need additional info?? We are here to help!

<details>
  <summary> Contact us!</summary>

The [Azure API Review Board](https://aka.ms/azapi) is dedicated to helping you create amazing APIs. You can read about our mission and learn more about our process on our [wiki](https://aka.ms/azapi).
* 💬 [Teams Channel](https://teams.microsoft.com/l/channel/19%3a3ebb18fded0e47938f998e196a52952f%40thread.tacv2/General?groupId=1a10b50c-e870-4fe0-8483-bf5542a8d2d8&tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47)
* 💌 [email](mailto://azureapirbcore@microsoft.com)

</details>

<details>
  <summary>Click here for links to tools, specs, guidelines & other good stuff</summary>

### Tooling
 * [Open API validation tools](https://aka.ms/swaggertools) were run on this PR. Go here to see [how to fix errors](https://aka.ms/ci-fix)
 * [Spectral Linting](https://aka.ms/style)
 * [Open API Hub](https://aka.ms/openapiportal)

### Guidelines & Specifications
 * [Azure REST API Guidelines](https://aka.ms/guidelines)
 * [OpenAPI Style Guidelines](https://aka.ms/style)
 * [Azure Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy)

### Helpful Links
 * [Azure DevTools Wiki](https://aka.ms/azapi)

</details>

### CODE REVIEW

File: specification/communication/data-plane/MediaComposition/preview/2022-07-16-preview/mediacomposition.json [hunk 1]
```diff
@@ -20,18 +20,15 @@
           {
             "in": "path",
             "name": "mediaCompositionId",
-            "description": "The media composition id",
+            "description": "",
```
Comment (DominikMe) [inline specification/communication/data-plane/MediaComposition/preview/2022-07-16-preview/mediacomposition.json:23]: How did we lose the description?

File: specification/communication/data-plane/MediaComposition/preview/2022-07-16-preview/mediacomposition.json [hunk 2]
```diff
@@ -20,18 +20,15 @@
...
             "required": true,
-            "type": "string",
-            "pattern": "^[0-9a-zA-Z-]+$",
-            "maxLength": 256
+            "type": "string"
```
Comment (DominikMe) [inline specification/communication/data-plane/MediaComposition/preview/2022-07-16-preview/mediacomposition.json:25]: Did we lose the other constraints by accident or on purpose?

File: specification/communication/data-plane/MediaComposition/preview/2022-07-16-preview/mediacomposition.json [hunk 3]
```diff
@@ -41,17 +38,46 @@
...
+            "schema": {
+              "$ref": "#/definitions/CommunicationErrorResponse"
+            },
+            "x-ms-error-response": true
+          },
           "default": {
             "description": "Failure",
             "schema": {
               "$ref": "#/definitions/CommunicationErrorResponse"
             },
```
Comment (DominikMe) [inline specification/communication/data-plane/MediaComposition/preview/2022-07-16-preview/mediacomposition.json:80]: We should keep returning the x-ms-error-code response header

File: specification/communication/data-plane/MediaComposition/preview/2022-07-16-preview/mediacomposition.json [hunk 4]
```diff
@@ -134,7 +185,7 @@
         "summary": "Updates an existing media composition",
         "operationId": "MediaComposition_Update",
         "consumes": [
-          "application/merge-patch+json"
+          "application/json"
```
Comment (DominikMe) [inline specification/communication/data-plane/MediaComposition/preview/2022-07-16-preview/mediacomposition.json:188]: This should still be `application/merge-patch+json`, why did this get changed?

**Review by DominikMe — COMMENTED** [review]
It looks like there are some regressions in the swagger, can we investigate what happened?

**Comment by DominikMe** [comment]
@anuchandy, @tg-msft, @johanste Our service isn't deployed yet and has never seen any customers - how should we treat the breaking changes / new api-version CI violations?

**Review by DominikMe — APPROVED** [review]

**Review by anuchandy — APPROVED** [review]

**Pull request merged**